File: OCH_Tao/tessoku-book/A12.js
const [[, K], A] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n").map(x => x.split(" ").map(y => parseInt(y, 10)));
let L = 0, R = 1e9;
while (L < R) {
  const M = Math.floor((L + R) / 2);
  let S = 0;
  A.forEach(x => S += Math.floor(M / x));
  if (K <= S) {
    R = M;
  } else {
    L = M + 1;
  }
}
console.log(L);

File: OCH_Tao/tessoku-book/A12.py
N, K = map(int, input().split())
A = list(map(int, input().split()))
L = 1
R = int(1e9)
while L < R:
  M = (L+R)//2
  S = 0
  for i in range(N):
    S += M//A[i]
  if K <= S:
    R = M
  else:
    L = M+1
print(L)
